dcore/state.py

```diff
@@ -75,32 +81,58 @@ def dump(self):
             self.output.dumplist = self.fieldlist
 
         funcs = self.xn.split()
+        field_dict = {name: func for (name, func) in zip(self.fieldlist, funcs)}
         to_dump = []
-
-        for name, f in zip(self.fieldlist, funcs):
+        for name, f in field_dict.iteritems():
             if name in self.output.dumplist:
                 to_dump.append(f)
             f.rename(name=name)
 
-        for field, meanfield in zip(funcs, self.output.meanfields):
+        steady_state_dump_err = self.output.steady_state_dump_err
+        for field in self.fieldlist:
+            steady_state_dump_err.setdefault(field, False)
+        for name, f, f_init in zip(self.fieldlist, funcs, self.x_init.split()):
+            if steady_state_dump_err[name]:
+                err = Function(f.function_space(), name=name+'err').assign(f-f_init)
+                field_dict[name+"err"] = err
+                self.diagnostics.register(name+"err")
+                to_dump.append(err)
+
+        meanfields = self.output.meanfields
```

Comment:
The meanfields and steady_state_dump_err loops can be combined if/when we write more general diagnostic fields - they are both just simply taking the difference of 2 fields.